### ISSUE TITLE
[a11y]: unable to put the heading of TableContainer at appropriate level

### ISSUE THREAD

**Issue by tshimizu-ibm** [carbon-design-system/carbon#18835]
### Package

@carbon/react

### Browser

Chrome

### Operating System

Windows

### Package version

1.76.0

### React version

18.3.1

### Automated testing tool and ruleset

IBM Accessibility Requirements 1.3.1 Info and Relationships

### Description

While I'm trying to make our product to meet IBM a11y requirement 1.3.1 https://www.ibm.com/able/requirements/requirements/?version=v7_3#1_3_1, I found `TableContainer` of `DataTable` contains `<h4>` hardcoded in [https://github.com/carbon-design-system/carbon/blob/fe90ac8ddb8364e229ecb3ef8037e3[…]3ee6/packages/react/src/components/DataTable/TableContainer.tsx](https://github.com/carbon-design-system/carbon/blob/fe90ac8ddb8364e229ecb3ef8037e368466c3ee6/packages/react/src/components/DataTable/TableContainer.tsx#L69)
so we cannot align the table with any content at the other levels semantically.

It should be Carbon's `<Heading>` component instead of `<h4>` tag so that we can control the level by wrapping the container with `<Section>` component, or a prop to specify the heading level should be provided.

### Reproduction/example

https://stackblitz.com/edit/github-k7vqxnzp?file=src%2FApp.tsx

### Steps to reproduce

1. Open the example app in Chrome
2. Open Chrome Dev Tools, or inspect TableContainer's heading in the preview from context menu
3. Find TableContainer's heading is marked as `<h4>` in Elements tab even though the app does not contain `<h3>` heading.
In the example app, it should be `<h2>` semantically as it's wrapped by `<section>` under `<h1>`.

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
